segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/ContainerRecoveryUtils.java

```diff
@@ -0,0 +1,210 @@
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.stream.Collectors;
+
+import static io.pravega.shared.NameUtils.getMetadataSegmentName;
+
+/**
+ * Utility methods for container recovery.
+ */
+@Slf4j
+public class ContainerRecoveryUtils {
+    private static final Duration TIMEOUT = Duration.ofSeconds(30);
+
+    /**
+     * This method lists the segments from the given storage instance. It then registers all segments except Attribute
+     * segments to the container metadata segment(s).
+     * {@link DebugStreamSegmentContainer} instance(s) are provided to this method which can have some segments already present
+     * in their respective container metadata segment(s). After the method successfully completes, only the segments which
+     * existed in the {@link Storage} will remain in the container metadata. All segments which only existed in the container
+     * metadata or which existed in both container metadata and the storage but with different lengths and/or sealed status,
+     * will be deleted from the container metadata. If the method fails while execution, appropriate exception is thrown.
+     * All segments from the storage are listed one by one, then mapped to their corresponding {@link DebugStreamSegmentContainer}
+     * instances for registering them to container metadata segment.
+     * @param storage                           A {@link Storage} instance that will be used to list segments from.
+     * @param debugStreamSegmentContainers      A Map of Container Ids to {@link DebugStreamSegmentContainer} instances
+     *                                          representing the containers that will be recovered.
+     * @param executorService                   A thread pool for execution.
+     * @throws InterruptedException             Required for Futures.get()
+     * @throws ExecutionException               Required for Futures.get()
+     * @throws TimeoutException                 Required for Futures.get()
+     * @throws IOException                      Requited for Storage.listSegments()
+     */
+    public static void recoverAllSegments(Storage storage, Map<Integer, DebugStreamSegmentContainer> debugStreamSegmentContainers,
+                                          ExecutorService executorService) throws InterruptedException, ExecutionException,
+            TimeoutException, IOException {
+        Preconditions.checkNotNull(storage);
+        Preconditions.checkNotNull(executorService);
+        Preconditions.checkNotNull(debugStreamSegmentContainers);
+        Preconditions.checkArgument(debugStreamSegmentContainers.size() > 0, "There should be at least one " +
```

Comment:
Loop through all the given containers and verify that they are all there (i.e., no duplicate IDs and that all of the expected Ids are provided). If you get 4 containers, you should expect ids 0 through 3.

Reply:
You can do this as a helper method and use a `HashSet`

Reply:
Done. Implemented another method.